File: examples-sdk/typescript/002-basic-vilapp-gateway.ts
#!/usr/bin/env tsx
// 002-basic-vilapp-gateway — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 002-basic-vilapp-gateway.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("vil-app-gateway", 3081);
const gw = new ServiceProcess("gw");
gw.endpoint("POST", "/trigger", "trigger_handler");
server.service(gw);
server.compile();

File: examples-sdk/python/002-basic-vilapp-gateway.py
#!/usr/bin/env python3
"""002-basic-vilapp-gateway — Python SDK equivalent
Compile: vil compile --from python --input 002-basic-vilapp-gateway.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("vil-app-gateway", port=3081)
gw = server.service_process("gw")
gw.endpoint("POST", "/trigger", "trigger_handler")
server.compile()
